mitodl/edx-sysadmin
# build: Django3.2 upgrade

#### Pre-Flight checklist


#### What are the relevant tickets?
#56 

**Reviewer Note:** 🔴 
In this PR I've used some other branch edx-platform than master to run the tests on as you can see [here](https://github.com/mitodl/edx-sysadmin/pull/57/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR31). The reason I'm doing that is that we have been using a test configuration `test_static_optimized`(Ref: https://github.com/mitodl/edx-sysadmin/blob/main/run_devstack_integration_tests.sh#L15) to generate our assets to test some of our functionality which now breaks because of import errors that are still not fixed Open edX's master. Although this is just the test config and doesn't really affect the actual functionality of the plugin.
I only did the above so it's easy for the reviewer to see the details of the passing tests. I'll set the platform's branch back to master upon merging this after the review/approval because in either case, the fix will be going to the master branch of the platform.


#### What's this PR do?
- Upgrades the package to use Django version 3.2 which is the same as edX.
- In the process of doing the above I've also run codemods edX's from [Open edX's repo-tools](https://github.com/edx/repo-tools/tree/master/edx_repo_tools/codemods/django3)
- Updated the tox files, requirements as of Django3.2
- Fixes/Ignores the warnings

#### How should this be manually tested?
- First off check that the tests are passing
- Following the instructions in the repo readme, install the sysadmin plugin
- Verify that you are able to access the sysadmin and able to test the functionality e.g. `Users Creation, Course Deletion, Course Imports and Git Logs`.

#### Where should the reviewer start?
Plugin installation

#### Screenshots (if appropriate)
<img width="1605" alt="Screenshot 2021-11-23 at 5 28 15 PM" src="https://user-images.githubusercontent.com/34372316/143024277-17da4fdc-ce60-45f9-8462-a0776f4b145c.png">
<img width="1537" alt="Screenshot 2021-11-23 at 5 28 44 PM" src="https://user-images.githubusercontent.com/34372316/143024294-7e2bdbf6-dcd0-47d9-82f8-b02233cc8321.png">
<img width="1277" alt="Screenshot 2021-11-23 at 5 28 58 PM" src="https://user-images.githubusercontent.com/34372316/143024299-7c45ab9f-dac1-431a-bae7-d4fed21f1a6e.png">
<img width="1352" alt="Screenshot 2021-11-23 at 5 29 11 PM" src="https://user-images.githubusercontent.com/34372316/143024301-2bb2d580-460e-4451-8abf-a4c7f7ec6286.png">
